### ISSUE TITLE
[Bug]: Could not create hypertable with the same name for more than one schema

### ISSUE THREAD

**Issue by kamil-di** [timescale/timescaledb#7150]
### What type of bug is this?

Unexpected error

### What subsystems and features are affected?

Compression

### What happened?

When trying to enable compression on the table with the same name but on different schema an error is returned.
		
	
	
            

### TimescaleDB version affected

2.15.0+

### PostgreSQL version used

15.0+

### What operating system did you use?

Ubuntu 22.04

### What installation method did you use?

Docker

### What platform did you run on?

Managed Service for TimescaleDB (MST/Aiven), Microsoft Azure Cloud, Other

### Relevant log output and stack trace

```bash
org.jkiss.dbeaver.model.sql.DBSQLException: SQL Error [P0003]: ERROR: query returned more than one row
  Hint: Make sure the query returns a single row, or use LIMIT 1.
  Where: PL/pgSQL function _timescaledb_functions.get_orderby_defaults(regclass,text[]) line 17 at SQL statement
SQL statement "SELECT  (SELECT string_agg(x, ', ') FROM jsonb_array_elements_text(seg_by->'clauses') t(x))::text,  seg_by->>'message',  (seg_by->>'confidence')::int FROM _timescaledb_functions.get_orderby_defaults(17111, coalesce($1, array[]::text[])) seg_by"
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:133)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:615)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$2(SQLQueryJob.java:506)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:192)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:525)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.extractData(SQLQueryJob.java:977)
	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.readData(SQLEditor.java:4176)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:123)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:192)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:121)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$ResultSetDataPumpJob.run(ResultSetViewer.java:5160)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:117)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

### How can we reproduce the bug?

```bash
CREATE SCHEMA "schema1";
CREATE TABLE "schema1"."completed_span_new"
(
	start_time        timestamptz not null,
	end_time          timestamptz not null,
	parent_start_time timestamptz,
	asset_id          bigint      not null,
	level             int         not null,
	program_name      text        not null,
	duration          interval generated always as (greatest(end_time - start_time, '00:00:00'::interval)) stored
);
select create_hypertable('schema1.completed_span_new', 'start_time');
select add_retention_policy('schema1.completed_span_new', INTERVAL '90 days');


CREATE SCHEMA "schema2";
CREATE TABLE "schema2"."completed_span_new"
(
	start_time        timestamptz not null,
	end_time          timestamptz not null,
	parent_start_time timestamptz,
	asset_id          bigint      not null,
	level             int         not null,
	program_name      text        not null,
	duration          interval generated always as (greatest(end_time - start_time, '00:00:00'::interval)) stored
);
select create_hypertable('schema2.completed_span_new', 'start_time');
select add_retention_policy('schema2.completed_span_new', INTERVAL '90 days');


ALTER TABLE "schema1"."completed_span_new"
	SET (
		timescaledb.compress,
		timescaledb.compress_segmentby = 'asset_id, level'
		);

ALTER TABLE "schema2"."completed_span_new"
	SET (
		timescaledb.compress,
		timescaledb.compress_segmentby = 'asset_id, level'
		);
```

**Comment by fabriziomello** [comment]
@kamil-di thanks for reporting. I was able to reproduce the issue:
```sql
92 (leader) postgres=# ALTER TABLE "schema1"."completed_span_new"
        SET (
                timescaledb.compress,
                timescaledb.compress_segmentby = 'asset_id, level'
                );
ERROR:  P0003: query returned more than one row
HINT:  Make sure the query returns a single row, or use LIMIT 1.
CONTEXT:  PL/pgSQL function _timescaledb_functions.get_orderby_defaults(regclass,text[]) line 17 at SQL statement
SQL statement "SELECT  (SELECT string_agg(x, ', ') FROM jsonb_array_elements_text(seg_by->'clauses') t(x))::text,  seg_by->>'message',  (seg_by->>'confidence')::int FROM _timescaledb_functions.get_orderby_defaults(20800, coalesce($1, array[]::text[])) seg_by"
LOCATION:  exec_stmt_execsql, pl_exec.c:4410
```

The problem is with the function to get the default `ORDER BY` for compression. Till we provide a fix an workaround you can use is something like this:
```sql
92 (leader) postgres=# SHOW timescaledb.compression_orderby_default_function ;
 timescaledb.compression_orderby_default_function 
--------------------------------------------------
 _timescaledb_functions.get_orderby_defaults
(1 row)

92 (leader) postgres=# SET timescaledb.compression_orderby_default_function = '';
SET
92 (leader) postgres=# ALTER TABLE "schema1"."completed_span_new"
        SET (
                timescaledb.compress,
                timescaledb.compress_segmentby = 'asset_id, level'
                );
ALTER TABLE
```

**Comment by fabriziomello** [comment]
@kamil-di actually this issue was already fixed in #7064 and will be available in the upcoming [2.16.0](https://github.com/timescale/timescaledb/milestone/86) release
